peerconnection.go

```diff
@@ -1138,8 +1138,8 @@ func (pc *PeerConnection) AddTrack(track *Track) (*RTPSender, error) {
 	var transceiver *RTPTransceiver
 	for _, t := range pc.rtpTransceivers {
 		if !t.stopped &&
-			// t.Sender == nil && // TODO: check that the sender has never sent
-			t.Sender.track == nil &&
```

Comment:
Initially, I thought that `RTPSender.track` cannot be nil, as it's validated in `PeerConnection.NewRTPSender`:

https://github.com/pions/webrtc/blob/f46ebf7effae3f373aa8fe8e0aa7c279572a49ad/rtpsender.go#L24-L27

But then I have double-checked the code and looks like, theoretically, track CAN be nil-ed:

https://github.com/pions/webrtc/blob/f46ebf7effae3f373aa8fe8e0aa7c279572a49ad/rtptransceiver.go#L19-L20

This method is called only once and only if we've found a matching transceiver in this loop (just few lines down the code):

https://github.com/pions/webrtc/blob/f46ebf7effae3f373aa8fe8e0aa7c279572a49ad/peerconnection.go#L1149-L1153

Would be correct to assume, that ever adding a nil track does not make any sense? Probably, it's even dangerous? Even more - there's a (commented out for now) FIXME for `PeerConnection.RemoveTrack`, so I assume `AddTrack` should never accept a nil track.

So if that's correct ^^ I'd like to add a nil-check for `track` argument for this method.

Reply:
Or, probably, would be better to add this `track != nil` check right in `RTPTransceiver.setSendingTrack`

Reply:
Yea this makes sense to me! We should never have a `nil` Track.

`setSendingTrack` also will break things. If we change the track, it currently doesn't change all the state required.

I am going to work on adding/removing tracks after signaling is complete but currently I think things will break :/

Reply:
So not sure if `setSendingTrack` will change much (or will be replaced with something other), but since I'm on it right now - will add a nil check there, why not.